### ISSUE TITLE
Ruby "filetype" unknown

### ISSUE THREAD

**Issue by toddyamakawa** [wellle/context.vim#32]
Here's an example where I open a Ruby script and jump to the bottom of the file. I see a "filetype unknown" error
![image](https://user-images.githubusercontent.com/7848869/73551615-5683dc80-440c-11ea-80d9-fac1d9d545c8.png)

When I run :set ft I see:
![image](https://user-images.githubusercontent.com/7848869/73551648-67345280-440c-11ea-8be2-1a17ea6310ea.png)

(My statusline also prints the filetype in the bottom right corner, and it also clearly says "ruby")

**Comment by wellle** [comment]
I can't reproduce this. Can you share the ruby file and share the full output of `:messages` after you've seen that error?

And can you tell me which Vim/Neovim version you're using?

**Comment by toddyamakawa** [comment]
Here's what I see with vim --version:

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan 18 2020 20:39:48)
Included patches: 1-129

Huge version without GUI.  Features included (+) or not (-):
+acl               -farsi             -mouse_sysmouse    -tag_old_static
+arabic            +file_in_path      +mouse_urxvt       -tag_any_white
+autocmd           +find_in_path      +mouse_xterm       -tcl
+autochdir         +float             +multi_byte        +termguicolors
-autoservername    +folding           +multi_lang        +terminal
-balloon_eval      -footer            -mzscheme          +terminfo
+balloon_eval_term +fork()            +netbeans_intg     +termresponse
-browse            -gettext           +num64             +textobjects
++builtin_terms    -hangul_input      +packages          +textprop
+byte_offset       +iconv             +path_extra        +timers
+channel           +insert_expand     -perl              +title
+cindent           +job               +persistent_undo   -toolbar
-clientserver      +jumplist          +popupwin          +user_commands
-clipboard         +keymap            +postscript        +vartabs
+cmdline_compl     +lambda            +printer           +vertsplit
+cmdline_hist      +langmap           +profile           +virtualedit
+cmdline_info      +libcall           -python            +visual
+comments          +linebreak         -python3           +visualextra
+conceal           +lispindent        +quickfix          +viminfo
+cryptv            +listcmds          +reltime           +vreplace
+cscope            +localmap          +rightleft         +wildignore
+cursorbind        -lua               -ruby              +wildmenu
+cursorshape       +menu              +scrollbind        +windows
+dialog_con        +mksession         +signs             +writebackup
+diff              +modify_fname      +smartindent       -X11
+digraphs          +mouse             -sound             -xfontset
-dnd               -mouseshape        +spell             -xim
-ebcdic            +mouse_dec         +startuptime       -xpm
+emacs_tags        -mouse_gpm         +statusline        -xsmp
+eval              -mouse_jsbterm     -sun_workshop      -xterm_clipboard
+ex_extra          +mouse_netterm     +syntax            -xterm_save
+extra_search      +mouse_sgr         +tag_binary        
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -O2 -fno-strength-reduce -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L/usr/local/lib -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -ldl           

```

Here's what I see with :messages:
![image](https://user-images.githubusercontent.com/7848869/73662373-4e70aa80-4661-11ea-8f0e-685a2e179d5f.png)

Here's the Ruby script:
https://github.com/toddyamakawa/bin/blob/master/diff-csv.rb

**Comment by wellle** [comment]
Thank you, I can reproduce by using Vim 8.2 on that file. If you type `:messages`, do you see this error message too?
>E363: pattern uses more memory than 'maxmempattern'

For example try renaming the file to `diff.c`. Now there's some (broken) syntax highlighting which should work in the context popup window.

I actually got a similar comment here, but didn't manage to reproduce it yet: https://www.reddit.com/r/vim/comments/eudnd8/welllecontextvim_update_use_vim_popupsneovim/ffuz97l/

So to me it seems that the ruby syntax highlighting is too demanding on the context buffer. I'm not sure if there's much I can do about that 🤔 

(The issue does not show up for me with Neovim btw)

**Comment by toddyamakawa** [comment]
> Thank you, I can reproduce by using Vim 8.2 on that file. If you type `:messages`, do you see this error message too?
> 
> > E363: pattern uses more memory than 'maxmempattern'

I do see that on other Ruby files, not on this one though. Any idea what causes that?
On an older version of context.vim I don't have any problems with Ruby: 065e105

> I actually got a similar comment here, but didn't manage to reproduce it yet: https://www.reddit.com/r/vim/comments/eudnd8/welllecontextvim_update_use_vim_popupsneovim/ffuz97l/

That's my reddit comment! :)

I did also open this one for C:
https://github.com/wellle/context.vim/issues/30

> So to me it seems that the ruby syntax highlighting is too demanding on the context buffer. I'm not sure if there's much I can do about that 🤔

When I first downloaded the plugin, I have any issues with Ruby. I have an older version of context.vim checked out that seems fine: 065e105

Anyways, great plugin! Thank you for making this :)

**Comment by wellle** [comment]
@toddyamakawa: I looked at it some more. To me it seems like the ruby syntax highlighting gets stuck on the border line (which is part of the popup buffer, so syntax highlighting gets applied there too). I tried making the border line a comment (by adding a `#` in the beginning), but that didn't help.

It seems like it's the non ASCII default border character `▬` (and other not ASCII characters). For example try putting this in your vimrc:
```vim
let g:context_border_char = '-'
```
This did fix it for me.

Let me know if that works for you!
